tiledb/sm/query/query_condition.cc

```diff
@@ -668,49 +695,130 @@ QueryCondition::apply_clause(
     case Datatype::STRING_UCS2:
     case Datatype::STRING_UCS4:
     default:
-      return {Status_QueryConditionError(
-                  "Cannot perform query comparison; Unsupported query "
-                  "conditional type on " +
-                  clause.field_name_),
-              nullopt};
+      throw std::runtime_error(
+          "QueryCondition::apply_ast_node: Cannot perform query comparison; "
+          "Unsupported query "
+          "conditional type on " +
+          node->get_field_name());
   }
 
-  return {Status::Ok(), nullopt};
+  return;
+}
+
+void QueryCondition::apply_tree(
+    const tdb_unique_ptr<ASTNode>& node,
+    const ArraySchema& array_schema,
+    uint64_t stride,
+    const std::vector<ResultCellSlab>& result_cell_slabs,
+    std::vector<uint8_t>& result_cell_bitmap) const {
+  // Case on the type of the tree node.
+  if (!node->is_expr()) {
+    // In the simple value node case, run the value evaluator function and
+    // return the result back.
+    apply_ast_node(
+        node, array_schema, stride, result_cell_slabs, result_cell_bitmap);
+  } else {
+    // For each child, declare a new result buffer, combine it all into the
+    // other one based on the combination op.
+    const QueryConditionCombinationOp& combination_op =
```

Comment:
In addition, in the case of variable initialization `auto x {5}` should be preferred over `auto x = 5`.  Though be careful when using this with actual constants, as the types they produce aren't always obvious.  Auto is nice when the type is clearly defined by, say, the return type of a function.  But my own preference is to state the type (stating intent) when that is important.

Reply:
I think using `const auto&` is better because it reflects the fact that the type is read-only and will not be modified. I've changed it as that locally but will keep this comment up if @ypatia or @lums658 want to comment more.

Reply:
It's fine with me